community/values/src/main/java/org/neo4j/values/storable/CoordinateReferenceSystem.java

```diff
@@ -147,4 +214,203 @@ public int hashCode()
+            assert p2.getCoordinateReferenceSystem().dimension == dimension;
+            return pythagoras( p1.coordinate(), p2.coordinate() );
+        }
+
+        @Override
+        public Pair<PointValue,PointValue> boundingBox( PointValue center, double distance )
+        {
+            assert center.getCoordinateReferenceSystem().dimension == dimension;
+            double[] coordinates = center.coordinate();
+            double[] min = new double[dimension];
+            double[] max = new double[dimension];
+            for ( int i = 0; i < dimension; i++ )
+            {
+                min[i] = coordinates[i] - distance;
+                max[i] = coordinates[i] + distance;
+            }
+            CoordinateReferenceSystem crs = center.getCoordinateReferenceSystem();
+            return Pair.of( Values.pointValue( crs, min ), Values.pointValue( crs, max ) );
+        }
+    }
+
+    private static class GeographicCalculator implements Calculator
+    {
+        private static final double EARTH_RADIUS_METERS = 6378140.0;
+        private static final double EXTENSION_FACTOR = 1.0001;
+        int dimension = 0;
+
+        GeographicCalculator( int dimension )
+        {
+            this.dimension = dimension;
+        }
+
+        @Override
+        public double distance( PointValue p1, PointValue p2 )
+        {
+            assert p1.getCoordinateReferenceSystem().dimension == dimension;
+            assert p2.getCoordinateReferenceSystem().dimension == dimension;
+            double[] c1Coord = p1.coordinate();
+            double[] c2Coord = p2.coordinate();
+            double[] c1 = new double[]{toRadians( c1Coord[0] ), toRadians( c1Coord[1] )};
+            double[] c2 = new double[]{toRadians( c2Coord[0] ), toRadians( c2Coord[1] )};
+            double dx = c2[0] - c1[0];
+            double dy = c2[1] - c1[1];
+            double alpha = pow( sin( dy / 2 ), 2.0 ) + cos( c1[1] ) * cos( c2[1] ) * pow( sin( dx / 2.0 ), 2.0 );
+            double greatCircleDistance = 2.0 * atan2( sqrt( alpha ), sqrt( 1 - alpha ) );
+            double distance2D = EARTH_RADIUS_METERS * greatCircleDistance;
+            if ( dimension > 2 )
+            {
+                double[] a = new double[dimension - 1];
+                double[] b = new double[dimension - 1];
+                a[0] = distance2D;
+                b[0] = 0.0;
+                for ( int i = 1; i < dimension - 1; i++ )
+                {
+                    a[i] = 0.0;
+                    b[i] = c1Coord[i + 1] - c2Coord[i + 1];
+                }
+                return pythagoras( a, b );
+            }
+            else
+            {
+                return distance2D;
+            }
+        }
+
+        @Override
+        // http://janmatuschek.de/LatitudeLongitudeBoundingCoordinates
+        // But calculating in degrees instead of radians to avoid rounding errors
+        public Pair<PointValue,PointValue> boundingBox( PointValue center, double distance )
```

Comment:
Did you also move the bounding box tests to reflect that this logic mas moved? Cannot find it in the changeset.

Reply:
That test was written in scala, and the best new location for it is in the Values module, which currently only has java tests, so I thought to leave it alone for the moment.